### DISCUSSION TITLE
Partial types and Graphcache

### DISCUSSION THREAD

**Discussion by thdxr** [urql-graphql/urql#2055] [Q&A] [answered]
This is may be a more general GraphQL question but struggling to find an answer

Suppose I have the following schema

```gql
type Author {
  id: ID!
  name: String!
  ...dozen other fields
}

type Post {
  id: ID!
  title: String!
  author: Author!
}
```

Is there a way to specify that when I do a `ListPosts` query, that the `author` field will only ever return the id field? That you cannot query deeper and get the full object?

One way I can think of doing this is by making another type called `AuthorRef` with just an ID field.  However I feel like this could confuse Graphcache

Any suggestions?

**Comment by kitten** [comment, marked as answer]
I'm not quite sure what the full intention here is, however, that wouldn't be making it a valid GraphQL schema.

Of course you could make a type, like, `AuthorEdge` that only returns an `id` field, however since you're asking whether it's possible to limit the fields per query operation, that's not possible. In fact, your schema is never limited by input query operations

This is especially important since you're only referring to it by name. There's no indication of what would limit this type. Now, if you mean you have a list of posts and once you've got those you can never reach the author, then that calls for special types.

It doesn't seem to me though like that's what you want. After all, it seems plausible that if you list posts, you may want to retrieve those posts' authors as well. Instead, I'm having a feeling that you maybe haven't set up your `Post` field resolvers to be able to retrieve `author` entirely?

What I'm getting at is that this feels like you've accidentally backed yourself into a corner in your GraphQL API implementation where you may be hesitant to implement a field resolver on `Post`?
